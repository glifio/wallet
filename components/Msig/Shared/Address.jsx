import React from 'react'
import PropTypes from 'prop-types'
import {
  Box,
  Glyph,
  CopyAddress,
  Label,
  IconLedger,
  ButtonClose,
  ButtonEdit
} from '@glif/react-components'
import { ADDRESS_PROPTYPE } from '../../../customPropTypes'

const Address = ({
  widthOverride,
  address,
  label,
  glyphAcronym,
  onRemoveSigner,
  onChangeSigner
}) => {
  return (
    <Box
      display='flex'
      alignItems='center'
      justifyContent='flex-start'
      color='core.darkgray'
      bg='background.messageHistory'
      height={6}
      px={2}
      mr={2}
      my={1}
      borderRadius={2}
      maxWidth={widthOverride ? 'none' : 12}
      width={widthOverride || 'auto'}
      minWidth={11}
    >
      {glyphAcronym ? (
        <Glyph
          mr={3}
          color='core.nearblack'
          acronym={glyphAcronym}
          size={5}
          border={0}
        />
      ) : (
        <Glyph
          justifyContent='flex-end'
          alignSelf='flex-end'
          mb='1px'
          mr={3}
          size={5}
          Icon={IconLedger}
          color='core.nearblack'
          bg='transparent'
          fill='#444'
          border={0}
          css='transform:translateY(-6px)'
        />
      )}
      <Box flexGrow='1'>
        <Label fontSize={1}>{label}</Label>
        <CopyAddress color='core.nearblack' address={address} />
      </Box>
      {onChangeSigner && (
        <ButtonEdit
          role='button'
          type='button'
          onClick={onChangeSigner}
          stroke='core.darkgray'
          aria-label='edit-signer'
        />
      )}
      {onRemoveSigner && (
        <ButtonClose
          fill='core.darkgray'
          role='button'
          type='button'
          onClick={onRemoveSigner}
          aria-label='remove-signer'
        />
      )}
    </Box>
  )
}

Address.propTypes = {
  address: ADDRESS_PROPTYPE,
  label: PropTypes.string.isRequired,
  glyphAcronym: PropTypes.string,
  widthOverride: PropTypes.string,
  onRemoveSigner: PropTypes.func,
  onChangeSigner: PropTypes.func
}

export default Address
