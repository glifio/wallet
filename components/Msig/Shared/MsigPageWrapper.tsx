import React, { ReactNode } from 'react'
import { node } from 'prop-types'
import {
  Box,
  LoadingScreen,
  Tooltip,
  BaseButton as ButtonLogout
} from '@glif/react-components'

import NavMenu from './NavMenu'
import { useMsig } from '../../../MsigProvider'
import MsgConfirmer from '../../../lib/confirm-message'
import { resetWallet } from '../../../utils/urlParams'

const MsigPageWrapper = ({ children }: { children: ReactNode }) => {
  const msig = useMsig()

  return (
    <Box
      display='flex'
      justifyContent='center'
      width='100%'
      minHeight='100vh'
      p={3}
    >
      <MsgConfirmer />
      <Box
        position='relative'
        display='flex'
        justifyContent='center'
        width='100%'
        p={3}
        // padding for logout button to ensure it never sits on top of the content
        paddingBottom={8}
      >
        {msig.loading && <LoadingScreen width='100%' />}
        {!msig.loading && (
          <Box display='flex' flexDirection='column' width='100%'>
            <NavMenu msigAddress={msig.Address} />
            {children}
          </Box>
        )}
        <ButtonLogout
          position='absolute'
          variant='secondary'
          bottom='0'
          left='0'
          m={5}
          display='flex'
          alignItems='center'
          justifyContent='space-between'
          css={`
            background-color: ${({ theme }) => theme.colors.core.secondary}00;
            &:hover {
              background-color: ${({ theme }) => theme.colors.core.secondary};
            }
          `}
          onClick={resetWallet}
        >
          Logout
          <Tooltip content='Logging out clears all your sensitive information from the browser and sends you back to the home page' />
        </ButtonLogout>
      </Box>
    </Box>
  )
}

MsigPageWrapper.propTypes = {
  children: node.isRequired
}

export default MsigPageWrapper
