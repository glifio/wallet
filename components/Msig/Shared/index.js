export * from './Headers'
export { default as GasCustomization } from './GasCustomization'
