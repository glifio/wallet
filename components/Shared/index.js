export { default as theme } from './theme'
export { default as ThemeProvider } from './ThemeProvider'

export { default as Container } from './Container'
export { default as Card } from './Card'
export { default as OnboardCard } from './Card/OnboardCard'
export { default as AccountCard } from './AccountCard'
export { default as AccountError } from './AccountCard/Error'
export { default as BalanceCard } from './BalanceCard'
export { default as ApproximationToggleBtn } from './BalanceCard/ApproximationToggleBtn'
export { default as Input } from './Input'
export { default as Address } from './Address'
export * from './Copy'
export { default as Button } from './Button'
export { default as BaseButton } from './Button/BaseButton'
export { default as Stepper } from './Stepper'
export { default as Glyph } from './Glyph'
export { default as Box } from './Box'
export { default as MessageHistoryTable } from './MessageHistoryTable'
export { default as FloatingContainer } from './FloatingContainer'
export { default as Loading } from './LoaderGlyph'
export { default as LoadingScreen } from './LoadingScreen'
export { default as NetworkSwitcherGlyph } from './NetworkSwitcherGlyph'
export { default as NodeConnectingGlyph } from './NodeConnected'
export { default as StepHeader } from './StepHeader'
export { default as ErrorView } from './Error'
export { default as Warning } from './Warning'
export * from './Link'
export * from './Menu'
export * from './Icons'
export * from './IconButtons'
export * from './MnemonicWord'

export * from './Layout'

export * from './Typography'
