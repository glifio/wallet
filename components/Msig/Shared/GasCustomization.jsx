import React, { useEffect, useRef, useState } from 'react'
import { func, string } from 'prop-types'
import { FilecoinNumber } from '@openworklabs/filecoin-number'
import { Box, Button, Text, IconLedger, Input, StyledATag } from '../../Shared'
import makeFriendlyBalance from '../../../utils/makeFriendlyBalance'
import useWallet from '../../../WalletProvider/useWallet'
import { FILECOIN_NUMBER_PROP } from '../../../customPropTypes'

// this is a weird hack to get tests to run in jest...
const NumberedInput = Input.Number

const GasCustomization = ({
  close,
  gasPrice,
  gasLimit,
  estimateGas,
  setGasPrice,
  setGasLimit,
  setEstimatedGas,
  value
}) => {
  const [gasPriceLocal, setGasPriceLocal] = useState(gasPrice)
  const [gasLimitLocal, setGasLimitLocal] = useState(gasLimit)
  const [estimatedGasFeeLocal, setEstimatedGasFeeLocal] = useState(
    new FilecoinNumber('122', 'attofil')
  )
  const [error, setError] = useState('')
  const [fetchedInitialEstimate, setFetchedInitialEstimate] = useState(false)

  useEffect(() => {
    const fetchInitialEstimate = async () => {
      try {
        const gas = await estimateGas(
          new FilecoinNumber('1', 'attofil'),
          gasLimit,
          value
        )
        setEstimatedGasFeeLocal(gas)
      } catch (err) {
        setError(err.message)
      }
    }

    if (!fetchedInitialEstimate) {
      setFetchedInitialEstimate(true)
      fetchInitialEstimate()
    }
  }, [
    fetchedInitialEstimate,
    setFetchedInitialEstimate,
    estimateGas,
    gasLimit,
    value
  ])

  const timeout = useRef(null)

  const onGasPriceInputChange = e => {
    const val = e.target.value
    setGasPriceLocal(new FilecoinNumber(val || '0', 'attofil'))
    clearTimeout(timeout.current)
    timeout.current = setTimeout(async () => {
      try {
        const gas = await estimateGas(
          new FilecoinNumber(val || '0', 'attofil'),
          gasLimit,
          value
        )
        setEstimatedGasFeeLocal(gas)
      } catch (err) {
        setError(err.message)
      }
    }, 500)
  }

  const onSubmit = e => {
    e.preventDefault()
    setGasPrice(gasPriceLocal)
    setGasLimit(gasLimitLocal)
    setEstimatedGas(estimatedGasFeeLocal)
    close()
  }

  const wallet = useWallet()

  return (
    <form onSubmit={onSubmit}>
      <Box p={3} bg='background.screen'>
        <Box display='flex' justifyContent='space-between' alignItems='center'>
          <Text my={3} pl={2} color='core.darkgray'>
            Transfers complete faster with a higher gas price.
          </Text>

          <StyledATag
            rel='noopener noreferrer'
            target='_blank'
            mt={0}
            color='core.primary'
            href='https://kb.myetherwallet.com/en/transactions/what-is-gas/'
          >
            What&rsquo;s this?
          </StyledATag>
        </Box>
        <Box>
          <Box mt={3}>
            <Input.Number
              top
              bottom={false}
              mt={2}
              m='0'
              denom='AttoFil'
              label='Gas Price'
              value={gasPriceLocal.toAttoFil()}
              onChange={onGasPriceInputChange}
            />
            <Box borderColor='background.screen'>
              <NumberedInput
                bottom
                top={false}
                m='0'
                denom='AttoFil'
                label='Gas Limit'
                value={gasLimitLocal.toAttoFil()}
                onChange={e =>
                  setGasLimitLocal(
                    new FilecoinNumber(e.target.value || '0', 'attofil')
                  )
                }
              />
            </Box>
          </Box>
        </Box>
      </Box>
      <Box
        display='flex'
        flexDirection='row'
        justifyContent='space-between'
        width='100%'
        p={3}
        border={0}
        bg='background.screen'
      >
        <Box>
          <Text margin={0}>Transaction Fee</Text>
          <Box display='flex' alignItems='center'>
            <Text margin={0} color='core.darkgray'>
              Paid via
            </Text>
            <IconLedger height='32px' />{' '}
            <Text margin={0} color='core.darkgray'>
              {makeFriendlyBalance(wallet.balance, 2, true)} FIL
            </Text>
          </Box>
        </Box>

        {/* <Box>
          <Text margin={0}>Transaction Fee</Text>
          <Text margin={0} color='core.darkgray'>
            Paid via <IconLedger /> {walletBalance} FIL
          </Text>
        </Box> */}
        <Text ml={2} color={error ? 'status.fail.background' : 'core.primary'}>
          {error
            ? 'Error fetching transaction fee'
            : `${estimatedGasFeeLocal.toAttoFil()} AttoFil`}
        </Text>
      </Box>

      <Box
        display='flex'
        flexDirection='row'
        justifyContent='space-between'
        bg='background.screen'
        p={3}
        borderBottomLeftRadius={3}
        borderBottomRightRadius={3}
      >
        <Button
          variant='secondary'
          title='Cancel'
          onClick={close}
          type='button'
        />
        <Button variant='primary' title='Save' disabled={error} type='submit' />
      </Box>
    </form>
  )
}

GasCustomization.propTypes = {
  close: func.isRequired,
  gasPrice: FILECOIN_NUMBER_PROP,
  gasLimit: FILECOIN_NUMBER_PROP,
  setGasPrice: func.isRequired,
  setGasLimit: func.isRequired,
  estimateGas: func.isRequired,
  setEstimatedGas: func.isRequired,
  value: string.isRequired
}

export default GasCustomization
