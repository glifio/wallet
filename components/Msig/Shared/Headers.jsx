import React from 'react'
import PropTypes from 'prop-types'
import { Box, Text, Glyph, StyledATag } from '../../Shared'
import {
  ADDRESS_PROPTYPE,
  FILECOIN_NUMBER_PROP
} from '../../../customPropTypes'
import makeFriendlyBalance from '../../../utils/makeFriendlyBalance'
import truncateAddress from '../../../utils/truncateAddress'

export const CardHeader = ({ address, msigBalance, signerBalance, msig }) => {
  return (
    <Box
      width='100%'
      p={3}
      border={0}
      borderTopRightRadius={3}
      borderTopLeftRadius={3}
      bg='core.primary'
      color='core.white'
    >
      <Box
        display='flex'
        flexDirection='row'
        alignItems='center'
        justifyContent='space-between'
      >
        <Box display='flex' flexDirection='row' alignItems='center'>
          <Glyph acronym='Ms' color='white' mr={3} />
          <Box display='flex' flexDirection='column' alignItems='flex-start'>
            <Text m={0}>From</Text>
            <Text m={0}>{truncateAddress(address)}</Text>
          </Box>
        </Box>
        {msig ? (
          <Box display='flex' flexDirection='row'>
            <Box
              display='flex'
              flexDirection='column'
              alignItems='flex-start'
              mr={3}
            >
              <Text m={0}>Ledger Balance</Text>
              <Text m={0}>
                {makeFriendlyBalance(signerBalance, 6, true)} FIL
              </Text>
            </Box>
            <Box display='flex' flexDirection='column' alignItems='flex-start'>
              <Text m={0}>Msig Balance</Text>
              <Text m={0}>{makeFriendlyBalance(msigBalance, 6, true)} FIL</Text>
            </Box>
          </Box>
        ) : (
          <Box display='flex' flexDirection='column' alignItems='flex-start'>
            <Text m={0}>Balance</Text>
            <Text m={0}>{makeFriendlyBalance(signerBalance, 6, true)} FIL</Text>
          </Box>
        )}
      </Box>
    </Box>
  )
}

CardHeader.propTypes = {
  address: ADDRESS_PROPTYPE,
  signerBalance: FILECOIN_NUMBER_PROP,
  msigBalance: FILECOIN_NUMBER_PROP,
  msig: PropTypes.bool
}

export const WithdrawHeaderText = ({ step }) => {
  let text = ''

  switch (step) {
    case 1:
      text =
        "First, please confirm the account you're sending from, and the recipient you want to send to."
      break
    case 2:
      text = "Next, please choose an amount of FIL you'd like to withdraw."
      break
    case 3:
      text =
        'Please review transaction fee details. Please note, the transaction fee is paid from the Filecoin wallet on your Ledger device, not your multisig wallet.'
      break
    case 4:
      text = 'Please review the transaction details.'
      break
    default:
      text = ''
  }

  return (
    <Box display='flex' flexDirection='column' alignItems='center'>
      <Text textAlign='center'>{text}</Text>
      <StyledATag
        width='auto'
        href='https://reading.supply/@glif/how-to-use-the-glif-vault-after-mainnet-launches-Td1ErO'
      >
        Click here for our guided Filecoin Vault tutorial.
      </StyledATag>
    </Box>
  )
}

WithdrawHeaderText.propTypes = {
  step: PropTypes.number.isRequired
}

export const ChangeOwnerHeaderText = ({ step }) => {
  let text = ''

  switch (step) {
    case 2:
      text =
        'Please input the new Filecoin address you want to be the owner of your multisig wallet.'
      break
    case 3:
      text =
        'Please review the transaction fee details. If the fee is too high, please come back and try again later.'
      break
    default:
      text = ''
  }
  return (
    <>
      <Text textAlign='center'>
        Your Ledger Address pays the transaction fee.
      </Text>
      <Text textAlign='center'>{text}</Text>
    </>
  )
}

ChangeOwnerHeaderText.propTypes = {
  step: PropTypes.number.isRequired
}

export const RemoveSignerHeaderText = ({ step }) => {
  let text = ''

  switch (step) {
    case 2:
      text =
        'Please review the transaction fee details. If the fee is too high, please come back and try again later.'
      break
    default:
      text = ''
  }
  return (
    <>
      <Text textAlign='center'>
        Your Ledger Address pays the transaction fee.
      </Text>
      <Text textAlign='center'>{text}</Text>
    </>
  )
}

RemoveSignerHeaderText.propTypes = {
  step: PropTypes.number.isRequired
}

export const AddSignerHeaderText = ({ step }) => {
  let text = ''

  switch (step) {
    case 2:
      text = 'Please enter the Filecoin address of the new signer.'
      break
    case 3:
      text =
        'Please review the transaction fee details. If the fee is too high, please come back and try again later.'
      break
    default:
      text = ''
  }
  return (
    <>
      <Text textAlign='center'>
        Your Ledger Address pays the transaction fee.
      </Text>
      <Text textAlign='center'>{text}</Text>
    </>
  )
}

AddSignerHeaderText.propTypes = {
  step: PropTypes.number.isRequired
}

export const AddRmSignerHeaderText = ({ method, step }) => {
  if (method === 5) return <AddSignerHeaderText step={step} />
  if (method === 6) return <RemoveSignerHeaderText step={step} />
}

AddRmSignerHeaderText.propTypes = {
  method: PropTypes.number.isRequired,
  step: PropTypes.number.isRequired
}
