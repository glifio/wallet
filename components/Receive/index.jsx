import React from 'react'
import styled from 'styled-components'
import { func } from 'prop-types'
import { ADDRESS_PROPTYPE } from '../../customPropTypes'
import { Address, Box, Glyph, Text, ButtonClose } from '../Shared'

const Wrapper = styled(Box).attrs(() => ({
  display: 'flex',
  flexDirection: 'column',
  border: 'none',
  p: 3,
  borderRadius: 2,
  borderWidth: 1,
  bg: 'background.screen'
}))``

const Receive = ({ address, close }) => {
  return (
    <Wrapper>
      <Box display='flex' alignItems='center' justifyContent='space-between'>
        <Box display='flex' alignItems='center'>
          <Glyph
            acronym='Rc'
            color='background.screen'
            borderColor='core.primary'
            backgroundColor='core.primary'
          />

          <Text color='core.primary' ml={2}>
            Receive Filecoin
          </Text>
        </Box>
        <Box display='flex' alignItems='center'>
          <ButtonClose ml={2} type='button' onClick={close} />
        </Box>
      </Box>
      <Text my={6}>
        To receive Filecoin, share your address with the sender
      </Text>
      <Address address={address} />
    </Wrapper>
  )
}

Receive.propTypes = {
  address: ADDRESS_PROPTYPE,
  close: func.isRequired
}

export default Receive
