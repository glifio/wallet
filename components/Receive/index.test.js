import React from 'react'
import { cleanup, render, act, fireEvent } from '@testing-library/react'
import Receive from '.'
import noop from '../../utils/noop'
import { ThemeProvider } from '../Shared'

describe('Receive', () => {
  afterEach(cleanup)

  test('it renders correctly', () => {
    const { container } = render(
      <ThemeProvider>
        <Receive
          address='t137sjdbgunloi7couiy4l5nc7pd6k2jmq32vizpy'
          close={noop}
        />
      </ThemeProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  test('it reveals the address when the user clicks reveal', () => {
    const { container, getByText } = render(
      <ThemeProvider>
        <Receive
          address='t137sjdbgunloi7couiy4l5nc7pd6k2jmq32vizpy'
          close={noop}
        />
      </ThemeProvider>
    )

    act(() => {
      fireEvent.click(getByText('Reveal'))
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
