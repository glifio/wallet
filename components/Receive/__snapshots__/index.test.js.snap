// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Receive it renders correctly 1`] = `
.c1 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  min-width: 0;
  box-sizing: border-box;
  border-width: 0.1875rem;
  border-style: solid;
  border-color: #5E26FF;
  background-color: #5E26FF;
  color: #EFF3FD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 48px;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  min-width: 0;
  box-sizing: border-box;
  padding-left: 4px;
  padding-right: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c13 {
  min-width: 0;
  box-sizing: border-box;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  font-family: "RT-Alias-Medium","system-ui","Segoe UI","Roboto",Helvetica;
  font-weight: 700;
  font-size: 1.5rem;
}

.c7 {
  width: 24;
  height: 24;
}

.c6 {
  outline: 0;
  border: 0;
  background: transparent;
  -webkit-transition: 0.24s ease-in-out;
  transition: 0.24s ease-in-out;
  cursor: pointer;
  display: inline-block;
  margin-left: 8px;
}

.c6:hover {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
}

.c5 {
  color: #5E26FF;
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  margin-left: 8px;
}

.c8 {
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  margin-top: 48px;
  margin-bottom: 48px;
}

.c11 {
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
}

.c12 {
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  margin-left: 4px;
  margin-right: 4px;
}

.c9 {
  min-width: 0;
  box-sizing: border-box;
  border: 1px solid;
  border-color: #999999;
  border-radius: 1px;
  padding-left: 16px;
  width: 640px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 70px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
}

.c14 {
  display: inline-block;
  height: 100%;
  outline: none;
  background-color: #E4EBFC;
  padding: 0;
  width: 64px;
  border: 0;
  border-left: 1px solid;
  border-color: #999999;
  border-top-right-radius: 1px;
  border-bottom-right-radius: 1px;
}

.c14:hover {
  cursor: pointer;
}

.c0 {
  min-width: 0;
  box-sizing: border-box;
  border: none;
  border-radius: 4px;
  border-width: 1px;
  padding: 16px;
  background-color: #EFF3FD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="c0"
  display="flex"
>
  <div
    class="c1"
    display="flex"
  >
    <div
      class="c2"
      display="flex"
    >
      <div
        class="c3"
        color="background.screen"
        display="flex"
        size="6"
      >
        <h3
          class="c4"
        >
          Rc
        </h3>
      </div>
      <p
        class="c5"
        color="core.primary"
        font-family="RT-Alias-Grotesk"
        font-size="3"
        font-weight="400"
      >
        Receive Filecoin
      </p>
    </div>
    <div
      class="c2"
      display="flex"
    >
      <button
        class="c6"
        display="inline-block"
        type="button"
      >
        <svg
          class="c7"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M19.7333 4.2669C19.3776 3.91103 18.801 3.91103 18.4453 4.2669L11.9994 10.7166L5.55552 4.26885C5.19986 3.91299 4.62323 3.91299 4.26757 4.26885C3.91191 4.62472 3.91191 5.20169 4.26757 5.55756L10.7115 12.0053L4.27793 18.4426C3.92228 18.7985 3.92228 19.3755 4.27793 19.7313C4.63359 20.0872 5.21022 20.0872 5.56587 19.7313L11.9994 13.294L18.435 19.7333C18.7906 20.0892 19.3672 20.0892 19.7229 19.7333C20.0786 19.3774 20.0786 18.8005 19.7229 18.4446L13.2874 12.0053L19.7333 5.5556C20.0889 5.19974 20.0889 4.62276 19.7333 4.2669Z"
            fill="#5E26FF"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
  <p
    class="c8"
    font-family="RT-Alias-Grotesk"
    font-size="3"
    font-weight="400"
  >
    To receive Filecoin, share your address with the sender
  </p>
  <div
    class="c9"
    width="14"
  >
    <div
      class="c10"
      display="flex"
    >
      <p
        class="c11"
        font-family="RT-Alias-Grotesk"
        font-size="3"
        font-weight="400"
      >
        t1
      </p>
      <p
        class="c12"
        font-family="RT-Alias-Grotesk"
        font-size="3"
        font-weight="400"
      >
        ●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●
      </p>
      <p
        class="c11"
        font-family="RT-Alias-Grotesk"
        font-size="3"
        font-weight="400"
      >
        izpy
      </p>
    </div>
    <div
      class="c13"
      display="flex"
      height="100%"
    >
      <button
        class="c14"
        font-family="RT-Alias-Grotesk"
        font-size="2"
        font-weight="400"
        width="7"
      >
        Copy
      </button>
      <button
        class="c14"
        font-family="RT-Alias-Grotesk"
        font-size="2"
        font-weight="400"
        width="7"
      >
        Reveal
      </button>
    </div>
  </div>
</div>
`;

exports[`Receive it reveals the address when the user clicks reveal 1`] = `
.c1 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  min-width: 0;
  box-sizing: border-box;
  border-width: 0.1875rem;
  border-style: solid;
  border-color: #5E26FF;
  background-color: #5E26FF;
  color: #EFF3FD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 48px;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c11 {
  min-width: 0;
  box-sizing: border-box;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  font-family: "RT-Alias-Medium","system-ui","Segoe UI","Roboto",Helvetica;
  font-weight: 700;
  font-size: 1.5rem;
}

.c7 {
  width: 24;
  height: 24;
}

.c6 {
  outline: 0;
  border: 0;
  background: transparent;
  -webkit-transition: 0.24s ease-in-out;
  transition: 0.24s ease-in-out;
  cursor: pointer;
  display: inline-block;
  margin-left: 8px;
}

.c6:hover {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
}

.c5 {
  color: #5E26FF;
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  margin-left: 8px;
}

.c8 {
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  margin-top: 48px;
  margin-bottom: 48px;
}

.c10 {
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  padding-left: 4px;
  padding-right: 8px;
}

.c9 {
  min-width: 0;
  box-sizing: border-box;
  border: 1px solid;
  border-color: #999999;
  border-radius: 1px;
  padding-left: 16px;
  width: 640px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 70px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
}

.c12 {
  display: inline-block;
  height: 100%;
  outline: none;
  background-color: #E4EBFC;
  padding: 0;
  width: 64px;
  border: 0;
  border-left: 1px solid;
  border-color: #999999;
  border-top-right-radius: 1px;
  border-bottom-right-radius: 1px;
}

.c12:hover {
  cursor: pointer;
}

.c0 {
  min-width: 0;
  box-sizing: border-box;
  border: none;
  border-radius: 4px;
  border-width: 1px;
  padding: 16px;
  background-color: #EFF3FD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="c0"
  display="flex"
>
  <div
    class="c1"
    display="flex"
  >
    <div
      class="c2"
      display="flex"
    >
      <div
        class="c3"
        color="background.screen"
        display="flex"
        size="6"
      >
        <h3
          class="c4"
        >
          Rc
        </h3>
      </div>
      <p
        class="c5"
        color="core.primary"
        font-family="RT-Alias-Grotesk"
        font-size="3"
        font-weight="400"
      >
        Receive Filecoin
      </p>
    </div>
    <div
      class="c2"
      display="flex"
    >
      <button
        class="c6"
        display="inline-block"
        type="button"
      >
        <svg
          class="c7"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M19.7333 4.2669C19.3776 3.91103 18.801 3.91103 18.4453 4.2669L11.9994 10.7166L5.55552 4.26885C5.19986 3.91299 4.62323 3.91299 4.26757 4.26885C3.91191 4.62472 3.91191 5.20169 4.26757 5.55756L10.7115 12.0053L4.27793 18.4426C3.92228 18.7985 3.92228 19.3755 4.27793 19.7313C4.63359 20.0872 5.21022 20.0872 5.56587 19.7313L11.9994 13.294L18.435 19.7333C18.7906 20.0892 19.3672 20.0892 19.7229 19.7333C20.0786 19.3774 20.0786 18.8005 19.7229 18.4446L13.2874 12.0053L19.7333 5.5556C20.0889 5.19974 20.0889 4.62276 19.7333 4.2669Z"
            fill="#5E26FF"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
  <p
    class="c8"
    font-family="RT-Alias-Grotesk"
    font-size="3"
    font-weight="400"
  >
    To receive Filecoin, share your address with the sender
  </p>
  <div
    class="c9"
    width="14"
  >
    <p
      class="c10"
      font-family="RT-Alias-Grotesk"
      font-size="3"
      font-weight="400"
    >
      t137sjdbgunloi7couiy4l5nc7pd6k2jmq32vizpy
    </p>
    <div
      class="c11"
      display="flex"
      height="100%"
    >
      <button
        class="c12"
        font-family="RT-Alias-Grotesk"
        font-size="2"
        font-weight="400"
        width="7"
      >
        Copy
      </button>
      <button
        class="c12"
        font-family="RT-Alias-Grotesk"
        font-size="2"
        font-weight="400"
        width="7"
      >
        Hide
      </button>
    </div>
  </div>
</div>
`;
